### ISSUE TITLE
`vic-machine upgrade` doesn't update the version reported by `docker info`

### ISSUE THREAD

**Issue by gigawhitlocks** [vmware/vic#4052]
Run `vic-machine create`, check version, looks something like:
```
~/go/src/github.com/vmware/vic master* 6s
❯ docker -H 192.168.218.149:2376 --tls info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: v0.8.0-85-5217a6a
```
(etc)

Rebuild with a new version number, e.g.:
```❯ bin/vic-machine-linux -v
vic-machine-linux version v0.8.0-89-5217a6a
```

Upgrade and see it succeed:

```~/go/src/github.com/vmware/vic master*
❯ /home/ian/go/src/github.com/vmware/vic/bin/vic-machine-linux upgrade --target 192.168.218.147 --user root  --password xxx --name ian-vch --thumbprint=xxx
Feb 27 2017 17:34:29.262-06:00 INFO  ### Upgrading VCH ####
Feb 27 2017 17:34:29.356-06:00 INFO  Validating target
Feb 27 2017 17:34:29.409-06:00 WARN  Disabling ha-host hostd.log collection (ServerFaultCode: Cannot complete the operation due to an incorrect request to the server.)
Feb 27 2017 17:34:29.409-06:00 INFO  
Feb 27 2017 17:34:29.409-06:00 INFO  VCH ID: VirtualMachine:7
Feb 27 2017 17:34:29.428-06:00 INFO  Uploading images for container
Feb 27 2017 17:34:29.428-06:00 INFO  	"/home/ian/go/src/github.com/vmware/vic/bin/bootstrap.iso"
Feb 27 2017 17:34:29.429-06:00 INFO  	"/home/ian/go/src/github.com/vmware/vic/bin/appliance.iso"
Feb 27 2017 17:34:39.461-06:00 INFO  Switching appliance iso to [datastore1] ian-vch/V0.8.0-89-5217A6A-appliance.iso
Feb 27 2017 17:34:39.478-06:00 INFO  Setting VM configuration
Feb 27 2017 17:34:39.814-06:00 INFO  Waiting for IP information
Feb 27 2017 17:34:39.834-06:00 INFO  Waiting for major appliance components to launch
Feb 27 2017 17:34:49.201-06:00 INFO  Obtained IP address for client interface: "192.168.218.149"
Feb 27 2017 17:34:49.201-06:00 INFO  Checking VCH connectivity with vSphere target
Feb 27 2017 17:34:49.272-06:00 INFO  vSphere API Test: https://192.168.218.147 vSphere API target responds as expected
Feb 27 2017 17:34:52.324-06:00 INFO  Deleting upgrade snapshot "upgrade for 85"
Feb 27 2017 17:34:52.445-06:00 INFO  Completed successfully
```

And check the version to see that it is not updated:
```
❯ docker -H 192.168.218.149:2376 --tls info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: v0.8.0-85-5217a6a
Storage Driver: vSphere Integrated Containers v0.8.0-85-5217a6a Backend Engine
VolumeStores: 
vSphere Integrated Containers v0.8.0-85-5217a6a Backend Engine: RUNNING
 VCH CPU limit: 2942 MHz
```

The version number of the Server Version etc should show 89 to match vic-machine.

**Comment by anchal-agrawal** [comment]
@emlin I've tagged this with medium priority, please re-assess if needed.

**Comment by gigawhitlocks** [comment]
After talking to @emlin it sounds like this is actually blocking my ability to test #3765 so I'm self-assigning and fixing this so that #3765 can move forward.

**Comment by emlin** [comment]
fix bye #4072
